src/commands/runs/ls.ts

```diff
@@ -0,0 +1,283 @@
+	if (providedActorNameOrId) {
+		const actorById = await client.actor(providedActorNameOrId).get();
+
+		if (actorById) {
+			return {
+				valid: true as const,
+				userFriendlyId: `${actorById.username}/${actorById.name}`,
+				id: actorById.id,
+			};
+		}
+
+		const actorByName = await client.actor(`${usernameOrId}/${providedActorNameOrId.toLowerCase()}`).get();
+
+		if (actorByName) {
+			return {
+				valid: true as const,
+				userFriendlyId: `${actorByName.username}/${actorByName.name}`,
+				id: actorByName.id,
+			};
+		}
+
+		return {
+			valid: false as const,
+			reason: `Actor with name or ID "${providedActorNameOrId}" was not found`,
+		};
+	}
+
+	if (localConfig.name) {
+		const actor = await client.actor(`${usernameOrId}/${localConfig.name}`).get();
+
+		if (!actor) {
+			return {
+				valid: false as const,
+				reason: `Actor with name "${localConfig.name}" was not found`,
+			};
+		}
+
+		return {
+			valid: true as const,
+			userFriendlyId: `${actor.username}/${actor.name}`,
+			id: actor.id,
+		};
+	}
+
+	return {
+		valid: false as const,
+		reason: 'Unable to detect what Actor to create a build for',
+	};
+}
+
+function prettyPrintStatus(status: string) {
+	switch (status) {
+		case 'READY':
+			return chalk.green('Ready');
+		case 'RUNNING':
+			return chalk.blue('Running');
+		case 'SUCCEEDED':
+			return chalk.green('Succeeded');
+		case 'FAILED':
+			return chalk.red('Failed');
+		case 'ABORTING':
+			return chalk.yellow('Aborting');
+		case 'ABORTED':
+			return chalk.red('Aborted');
+		case 'TIMING-OUT':
+			return chalk.yellow('Timing Out');
+		case 'TIMED-OUT':
+			return chalk.red('Timed Out');
+		default:
+			return chalk.gray(
+				(status as string)
+					.split('-')
+					.map((part) => part[0].toUpperCase() + part.slice(1).toLowerCase())
+					.join(' '),
+			);
+	}
+}
+
+const ShortDurationFormatter = new SapphireDurationFormatter({
+	[TimeTypes.Day]: {
+		DEFAULT: 'd',
+	},
+	[TimeTypes.Hour]: {
+		DEFAULT: 'h',
+	},
+	[TimeTypes.Minute]: {
+		DEFAULT: 'm',
+	},
+	[TimeTypes.Month]: {
+		DEFAULT: 'M',
+	},
+	[TimeTypes.Second]: {
+		DEFAULT: 's',
+	},
+	[TimeTypes.Week]: {
+		DEFAULT: 'w',
+	},
+	[TimeTypes.Year]: {
+		DEFAULT: 'y',
+	},
+});
+
+// END of TODO
+
+const multilineTimestampFormatter = new Timestamp(`YYYY-MM-DD[\n]HH:mm:ss`);
+const tableFactory = (compact = false) => {
+	const options: Record<string, unknown> = {
+		head: ['ID', 'Status', 'Results', 'Usage', 'Start Date', 'Took', 'Build Number', 'Origin'],
+		style: {
+			head: ['cyan', 'cyan', 'cyan', 'cyan', 'cyan', 'cyan', 'cyan', 'cyan'],
+			compact,
+		},
+		colAligns: ['middle', 'middle', 'middle', 'middle', 'middle', 'middle', 'middle', 'middle'],
+	};
+
+	if (compact) {
+		options.chars = {
+			'mid': '',
+			'left-mid': '',
+			'mid-mid': '',
+			'right-mid': '',
+			middle: ' ',
+			'top-mid': '─',
+			'bottom-mid': '─',
+		};
+	}
+
+	return new Table<[string, string, string, string, string, string, string, string]>(options);
```

Comment:
i wouldn't bother with the type argument here if it would mean repeating `string` N times, the method is not meant to be used by any user so there is little to no value in this